plugins/mcreator-core/blockly/mcreator_extensions.js

```diff
@@ -109,6 +109,18 @@ Blockly.Extensions.register('entity_list_provider',
         this.appendDummyInput().appendField(new FieldDataListSelector('entity'), 'entity');
     });
 
+Blockly.Extensions.register('custom_entity_with_entity_data_list_provider',

```

Comment:
I tried creating two different entities with data, but the data dropdown only shows the variables from the first entity

Reply:
So switching a `customentity` should update contents of `dataName` field via <kbd>callback</kbd> function, like data list selectors do this: https://github.com/MCreator/MCreator/blob/a512add80a3d109dff82c1c8578474130c75eb1a/plugins/mcreator-core/blockly/field_data_list_selector.js#L56

Reply:
https://github.com/MCreator/MCreator/pull/2007/commits/fc8ef86402a2e6a449d62dbc07dd8c6f4e70cca8